docs/design/2025-prometheus-mapping-component.md

```diff
@@ -0,0 +1,45 @@
+# Proposal: Add a component to perform label mapping efficiently
+
+* Author(s): Nicolas DUPEUX
+* Last updated: 19/11/2024
+* Original issue: https://github.com/grafana/alloy/pull/2025
+
+## Abstract
+
+Add a component to populate labels values based on a lookup table.
+
+## Problem
+
+Using `prometheus.relabel` to populate a label value based on another label value is inefficient as we have to have a rule block for each source label value.
+
+If we have 1k values to map, we'll have to execute 1k regex for each datapoint resulting in an algorithm complexity of O(n).
+
+## Proposal
+
+Replace regex computing by a lookup table. Algorithm complexity goes from O(n) to O(1)
```

Comment:
I guess `loki.mapping` or `discovery.mapping` could be useful too. Happy to leave it for future PRs though.